addons/textalog/nodes/dialogue_base_node/portraits_node/portraits_node.gd

```diff
@@ -40,19 +40,25 @@ func add_portrait(
 		emit_signal("portrait_added", character, portrait)
 		return
 	
-	# Remove previous node
+	
+	var _texture_rect:TextureRect
+	
+	# Use previous node as reference
 	if character in portraits:
+		_texture_rect = portraits.get(character, null)
 		remove_portrait(character)
 		emit_signal("portrait_changed", character, portrait)
-	
-	var _texture_rect:TextureRect = TextureRect.new()
-	connect("tree_exiting", _texture_rect, "queue_free")
+	# Create a new node as no previous node exists
+	else:
+		_texture_rect = TextureRect.new()
+		portraits[character] = _texture_rect
+
+	if not is_connected("tree_exiting", _texture_rect, "queue_free"):
+		connect("tree_exiting", _texture_rect, "queue_free")
```

Comment:
You're removing the portrait if the character exists in scene.

It should use the already existing one and verify if it's a valid instance.

```GDScript
if character in portraits:
	_texture_rect = portraits.get(character, null)
	emit_signal("portrait_changed", character, portrait)

if not is_instance_valid(_texture_rect):
	_texture_rect = TextureRect.new()
	portraits[character] = _texture_rect


if not is_connected("tree_exiting", _texture_rect, "queue_free"):
	connect("tree_exiting", _texture_rect, "queue_free")
```